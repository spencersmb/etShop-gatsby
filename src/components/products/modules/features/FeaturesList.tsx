import FeatureItem from '@components/products/modules/features/featureItem'
import { IFeatureItem } from '@et/types/Products'
import { device } from '@styles/global/breakpoints'
import { colors } from '@styles/global/colors'
import { GridFluid } from '@styles/global/cssGrid'
import { Sentinel } from '@styles/global/fonts'
import { svgs } from '@svg'
import { renderSvg } from '@utils/styleUtils'
import React from 'react'
import styled from 'styled-components'

interface IProps {
	features: IFeatureItem[]
}

const FeaturesList = (props: IProps) => {

	const { features } = props
	return (
		<Container>
			<TitleHeader data-testid={'title'}>
				Features
			</TitleHeader>
			<Features data-testid={'items'}>
				{features.map((feature, index) => (<FeatureItem index={index} key={index} {...feature}/>))}
			</Features>
			<Divider>
				{renderSvg(svgs.Divider1)}
			</Divider>
		</Container>
	)
}
const Features = styled.div`
	grid-column: 2 /4;
	position: relative;
	z-index: 2;
	align-items: flex-start;
	@media ${device.tablet} {
		grid-column: 3 / 13;
	}
	
	@media ${device.laptop} {
		grid-column: 2 / 14;
		display: grid;
		grid-template-columns: 1fr 1fr;
		grid-template-rows: 80px 1fr 80px 1fr;
		//flex-direction: row;
		//flex-wrap: wrap;
		//align-items: flex-start;
	}
		
`
const Divider = styled.div`
width: 100%;
position: absolute;
bottom:0;
left:0;
z-index: 1;
height: 350px;
overflow: hidden;
grid-column: 1 / -1;
	svg{
		width: 850px;
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
	}
	
	@media ${device.tablet} {
		svg{
			width: 1500px;
		}
	}
	
	@media ${device.laptop} {
		svg{
			width: 2020px;
		}
	}
	
		@media ${device.laptopL} {
		svg{
			width: 3020px;
		}
	}
	
`
const Container = styled(GridFluid)`
grid-row-gap: 0 !important;
padding: 60px 0 80px;
overflow: hidden;
position: relative;
z-index: 1;

	@media ${device.laptop} {
		margin-top: 80px;
		padding: 110px 0 150px;
	}
	@media ${device.laptopL} {
		padding: 190px 0 180px;
	}
`
const TitleHeader = styled.h5`
	font-weight: 500;
	${Sentinel.semiboldItalic};
	font-style: italic;
	position: relative;
	color: ${colors.primary.headline};
	grid-column: 2 /4;
	margin-bottom: 25px;
	font-size: 28px;
	z-index: 2;
	text-align: center;
	
	@media ${device.tablet} {
    font-size: 34px;
    font-weight: 500;
    font-style: italic;
    margin-bottom: 25px;
    text-align: center;
    grid-column: 2 / 14;
	}
	
	@media ${device.laptop} {
		color: #d6e0ea;
		left: 0;
		top: -30px;
		font-size: 190px;
    line-height: 190px;
    position: absolute;
	}
	@media ${device.laptopL} {
		top: 0;
		font-size: 248px;
    line-height: 248px;
	}
		
`
export default React.memo(FeaturesList)
