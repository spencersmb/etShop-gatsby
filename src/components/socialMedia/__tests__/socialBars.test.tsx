import FeaturesList from '@components/products/modules/features/FeaturesList'
import SocialMediaBars from '@components/socialMedia/socialBars'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const propsDefault = {
	bars: [
		{
			type: 'youtube-tuts'
		},
		{
			type: 'procreate-Instagram'
		}
	]
}
describe('Product Description', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<SocialMediaBars {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct # of items', () => {
		const modalRender = render(<SocialMediaBars {...propsDefault}/>)
		expect(modalRender.getAllByTestId('item').length).toEqual(propsDefault.bars.length)
	})
})
