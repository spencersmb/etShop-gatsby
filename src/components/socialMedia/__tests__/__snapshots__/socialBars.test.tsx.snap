// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Description renders correctly 1`] = `
Array [
  <div
    className="sc-bdVaJa jyvzgi"
    data-testid="item"
  >
    <div
      className="sc-bwzfXH sc-htpNat fHXyX"
      color="#fff"
      data-testid="icon"
      type="youtube-tuts"
    >
      <test-file-stub />
    </div>
    <div
      className="sc-ifAKCX eUWlWz"
    >
      <div
        className="sc-bZQynM bBieGp"
        data-testid="cat"
      >
        youtube
      </div>
      <h4
        className="sc-EHOje eDbdtz"
        data-testid="title"
      >
        Looking for free tutorials?
      </h4>
      <div
        className="sc-gzVnrw bxTmTy"
        data-testid="desc"
      >
        You can also find me every Tuesday sharing a new design, Procreate or lettering tutorial on YouTube
      </div>
    </div>
    <a
      className="sc-htoDjs jlYWml"
      data-testid="btn"
      href="https://youtube.com/everytues"
      rel="noreferrer"
      target="_blank"
    >
      watch
    </a>
  </div>,
  <div
    className="sc-bdVaJa jyvzgi"
    data-testid="item"
  >
    <div
      className="sc-bwzfXH sc-htpNat hTJhck"
      color="#fff"
      data-testid="icon"
      type="procreate-Instagram"
    >
      <test-file-stub />
    </div>
    <div
      className="sc-ifAKCX eUWlWz"
    >
      <div
        className="sc-bZQynM bBieGp"
        data-testid="cat"
      >
        Instagram
      </div>
      <h4
        className="sc-EHOje eDbdtz"
        data-testid="title"
      >
        See more Procreate artwork
      </h4>
      <div
        className="sc-gzVnrw bxTmTy"
        data-testid="desc"
      >
        To see more Procreate artwork using these brushes, find me on Instagram!
      </div>
    </div>
    <a
      className="sc-htoDjs jlYWml"
      data-testid="btn"
      href="https://instagram.com/everytuesday"
      rel="noreferrer"
      target="_blank"
    >
      Follow
    </a>
  </div>,
]
`;
